catalog/rocks/staging/manager.go

```diff
@@ -0,0 +1,111 @@
+package staging
+
+import (
+	"context"
+
+	"github.com/treeverse/lakefs/catalog/rocks"
+	"github.com/treeverse/lakefs/db"
+	"github.com/treeverse/lakefs/logging"
+)
+
+type Manager struct {
+	db  db.Database
+	log logging.Logger
+}
+
+func NewManager(db db.Database) *Manager {
+	return &Manager{
+		db:  db,
+		log: logging.Default().WithField("service_name", "postgres_staging_manager"),
+	}
+}
+
+func (p *Manager) GetEntry(ctx context.Context, st rocks.StagingToken, path rocks.Path) (*rocks.Entry, error) {
+	res, err := p.db.Transact(func(tx db.Tx) (interface{}, error) {
+		entry := &rocks.Entry{}
+		err := tx.Get(entry, "SELECT address, last_modified_date, size, checksum, metadata FROM staging_entries WHERE staging_token=$1 AND path=$2", st, path)
+		return entry, err
+	}, p.txOpts(ctx, db.ReadOnly())...)
+	if err != nil {
+		return nil, err
+	}
+	entry := res.(*rocks.Entry)
+	if entry.Address == "" {
+		return nil, nil
+	}
+	return entry, nil
+}
+
+func (p *Manager) SetEntry(ctx context.Context, st rocks.StagingToken, path rocks.Path, entry *rocks.Entry) error {
+	if entry == nil {
+		entry = &rocks.Entry{}
+	}
+	_, err := p.db.Transact(func(tx db.Tx) (interface{}, error) {
+		return tx.Exec("INSERT INTO staging_entries"+
+			"(staging_token, path, address, last_modified_date, size, checksum, metadata) "+
+			"VALUES ($1, $2,$3, $4, $5, $6, $7)"+
+			"ON CONFLICT (staging_token, path)"+
+			"DO UPDATE SET (staging_token, path, addre"+
+			"ss, last_modified_date, size, checksum, metadata) ="+
+			"(excluded.staging_token, excluded.path, excluded.address, excluded.last_modified_date, excluded.size, excluded.checksum, excluded.metadata)",
+			st, path, entry.Address, entry.LastModified, entry.Size, entry.ETag, entry.Metadata)
+	}, p.txOpts(ctx)...)
+	return err
+}
+
+func (p *Manager) DeleteEntry(ctx context.Context, st rocks.StagingToken, path rocks.Path) error {
+	_, err := p.db.Transact(func(tx db.Tx) (interface{}, error) {
+		return tx.Exec("DELETE FROM staging_entries WHERE staging_token=$1 AND path=$2", st, path)
+	}, p.txOpts(ctx)...)
+	return err
+}
+
+func (p *Manager) ListEntries(ctx context.Context, st rocks.StagingToken, from rocks.Path) (rocks.EntryIterator, error) {
+	return NewIterator(ctx, p, st, from), nil
+}
+
+type listEntriesResult struct {
+	entries  []*rocks.EntryRecord
+	hasNext  bool
+	nextPath rocks.Path
+}
+
+func (p *Manager) listEntries(ctx context.Context, st rocks.StagingToken, from rocks.Path, limit int) (*listEntriesResult, error) {
```

Comment:
I thought it will be the other way around - iterator will select the data, listing with pagination will use the iterator.
The layer above doesn't have access to list entries - it will re-implement this part using the iterator.